rust/src/io/commit/external_manifest.rs

```diff
@@ -46,11 +49,14 @@ pub trait ExternalManifestStore: std::fmt::Debug + Send + Sync {
     /// Get the manifest path for a given base_uri and version
     async fn get(&self, base_uri: &str, version: u64) -> Result<String>;
 
-    /// Get the latest version of a dataset at the base_uri
-    async fn get_latest_version(&self, base_uri: &str) -> Result<Option<u64>>;
+    /// Get the latest version of a dataset at the base_uri, and the path to the manifest
```

Comment:
Why do we need the path? Shouldn't it be uniquely determined by the version? Or is this just an optimization?

Reply:
This is an optimization to not have to `get(get_latest_version)`

Reply:
Got it. Let's specify that in the docs:
```suggestion
    /// Get the latest version of a dataset at the base_uri, and the path to the manifest.
    /// The path is provided as an optimization. The path is deterministic based on
    /// the version and the store should not customize it.
```

Reply:
done